great_expectations/expectations/expectation.py

```diff
@@ -336,6 +336,20 @@ def schema_extra(schema: Dict[str, Any], model: Type[Expectation]) -> None:
         default=None
     )
 
+    @override
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Expectation):
+            return False
+
+        self_dict = self.dict()
+        other_dict = other.dict()
+
+        for attr in ("notes", "meta"):
```

Comment:
This will change the objects in place. We'll need to make copies if you do this, eg
```
almost_equal = dict(self_dict, {"notes": None, "meta": None}) == dict(other_dict, {"notes": None, "meta": None})
```
I would do follow up checks on `notes` and `meta` to check their equality allowing for the equivalence of `None` and `{}`.